indra/newview/llviewertexturelist.cpp

```diff
@@ -1113,6 +1147,62 @@ F32 LLViewerTextureList::updateImagesCreateTextures(F32 max_time)
     return create_timer.getElapsedTimeF32();
 }
 
+void LLViewerTextureList::updateImagesNameTextures()
+{
+    LL_PROFILE_ZONE_SCOPED_CATEGORY_TEXTURE;
+    if (gGLManager.mIsDisabled) { return; }
+    static LLCachedControl<bool> debug_texture_label(gSavedSettings, "RenderDebugTextureLabel", false);
+    if (!debug_texture_label()) { return; }
+
+    static GLsizei max_length = 0;
+    if (max_length == 0) { glGetIntegerv(GL_MAX_LABEL_LENGTH, &max_length); }
+
+    auto it = mNameTextureList.begin();
+    while (it != mNameTextureList.end()) // For ALL textures needing names
+    {
+        if (it->mTex->hasGLTexture())
+        {
+            if(it->mTex->getTexName())
+            {
+                it->mTex->setGLObjectLabel(it->mPrefix, true);
+                it = mNameTextureList.erase(it); // Assume no rename needed
+            }
+            else
+            {
+                ++it; // Not ready
+            }
+        }
+        else
+        {
+            ++it; // Not ready
+        }
+    }
+}
+
+void LLViewerTextureList::labelAll()
+{
+    LL_PROFILE_ZONE_SCOPED_CATEGORY_TEXTURE;
+    static LLCachedControl<bool> debug_texture_label(gSavedSettings, "RenderDebugTextureLabel", false);
+    if (!debug_texture_label()) { return; }
+
+    static const std::string local_prefix = "lltexlocal";
+    static const std::string other_prefix = "lltexother";
+
+    std::string label;
+    bool error;
+    for (image_list_t::iterator it = mImageList.begin(); it != mImageList.end(); ++it)
+    {
+        LLViewerFetchedTexture* image = *it;
```

Comment:
probably would prefer
```suggestion
    for (LLViewerFetchedTexture * image : mImageList)
    {
```

Reply:
Yea that wasn't necessary for this loop. Always good to simplify.